goldeneye/src/main/java/co/infinum/goldeneye/GoldenEye.kt

```diff
@@ -184,31 +186,48 @@ interface GoldenEye {
         fun withAdvancedFeatures() = apply { this.advancedFeaturesEnabled = true }
 
         /**
-         * Forces the use of [GoldenEye1Impl] regarding of the device.
+         * Manually select Camera API version.
          *
-         * CAUTION: Video recording does not work for some newer devices that are supposed
-         * to be using Camera2 instead.
+         * It might be useful to force Camera1 API if application does not use video recording feature
+         * because it is more consistent when taking pictures with [co.infinum.goldeneye.models.FlashMode.ON].
+         *
+         * CAUTION: Video recording on newer Android devices can crash when using Camera1 API!
+         *
+         * @throws IllegalArgumentException when trying to force Camera2 API on devices older than Lollipop.
+         */
+        @Throws(IllegalArgumentException::class)
+        fun setCameraApi(cameraApi: CameraApi): GoldenEye.Builder {
+            if (cameraApi == CameraApi.CAMERA2 && Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+                throw IllegalArgumentException("Camera2 API is available from SDK 21.")
+            }
+
+            return apply { this.cameraApi = cameraApi }
+        }
+
+        /**
+         * Behavior can be overridden by using [setCameraApi] method.
+         *
+         * @return true if GoldenEye will use Camera2 API, otherwise false.
          */
-        fun forceCamera1Api() = apply { this.forceCamera1Api = true }
+        fun shouldUseCamera2Api() =
```

Comment:
@ikocijan @mariciv @dmarusic35 should this method be here or exposed somewhere else? I am not big fan of this method inside `Builder`.

Reply:
I agree it doesn't belong in Builder. can it be some static method outside of builder?